src/stepfunctions/steps/sagemaker.py

```diff
@@ -115,14 +118,20 @@ def __init__(self, state_id, estimator, job_name, data=None, hyperparameters=Non
         if isinstance(job_name, Placeholder):
             parameters['TrainingJobName'] = job_name
 
-        if output_path is not None:
-            parameters['OutputDataConfig']['S3OutputPath'] = output_path
+        if output_data_config_path is not None:
+            parameters['OutputDataConfig']['S3OutputPath'] = output_data_config_path
 
         if data is not None and is_data_placeholder:
             # Replace the 'S3Uri' key with one that supports JSONpath value.
             # Support for uri str only: The list will only contain 1 element
-            temp_data_uri = parameters['InputDataConfig'][0]['DataSource']['S3DataSource'].pop('S3Uri', None)
-            parameters['InputDataConfig'][0]['DataSource']['S3DataSource']['S3Uri.$'] = temp_data_uri
+            try:
+                data_uri = parameters['InputDataConfig'][0]['DataSource']['S3DataSource'].pop('S3Uri', None)
+                parameters['InputDataConfig'][0]['DataSource']['S3DataSource']['S3Uri.$'] = data_uri
+            except KeyError as error:
```

Comment:
This could also result in an `IndexError` if `InputDataConfig` is empty. My concern was if `training_config` always returns a structure like this. If it's guaranteed getting here actually is exceptional and we shouldn't swallow the error unless we can handle it somehow.

Reply:
It is guaranteed as long as the structure returned by [sagemaker.workflow.airflow.training_config](https://sagemaker.readthedocs.io/en/stable/workflows/airflow/sagemaker.workflow.airflow.html) stays the same.

Reply:
I'll remove the try catch so that the error is not swallowed when the structure is changed